src/Compilers/Core/Portable/StrongName/PortableStrongNameProvider.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Cci;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class PortableStrongNameProvider : StrongNameProvider
+    {
+        private readonly ImmutableArray<string> _keyFileSearchPaths;
+        internal StrongNameFileSystem FileSystem { get; private set; }
+
+        public PortableStrongNameProvider(ImmutableArray<string> keySearchPaths = default, StrongNameFileSystem ioOp = default)
+        {
+            FileSystem = ioOp ?? new StrongNameFileSystem();
+            _keyFileSearchPaths = keySearchPaths.NullToEmpty();
+        }
+
+        public override bool Equals(object other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (other.GetType() != this.GetType())
+            {
+                return false;
+            }
+
+            return Capability == (other as PortableStrongNameProvider).Capability;
+        }
```

Comment:
Can be simplified to:
```
var otherProvider = other as PortableStrongNameProvider;
return otherProvider != null && Capability == otherProvider.Capability;
```

Reply:
I'm not sure why I'd use the `as` operator here. The correct implementation of this method is the `other.GetType() != this.GetType()` enforcing that they are *exactly* the same type, not allowing other to be a subtype of PortableStrongNameProvider.

With that in mind, I know that the cast can't fail.

Reply:
The exact type comparison is not needed because `PortableStrongNameProvider` is `sealed` and cannot have subtypes.

Reply:
Good point.  I can simplify this to `return other.GetType() == GetType()`.

Reply:
Good point.  I've changed this to:

```
            if (other is PortableStrongNameProvider o)
            {
                return _keyFileSearchPaths == o._keyFileSearchPaths && FileSystem == o.FileSystem;
            }
            else
            {
                return false;
            }
```